core/trigger.class.inc.php

```diff
@@ -546,6 +546,40 @@ public static function Init()
 		MetaModel::Init_SetZListItems('standard_search', array('description', 'target_class')); // Criteria of the std search form
 	}
 
+	/**
+	 * Activate trigger based on attribute list given instead of changed attributes
+	 *
+	 * @param array $aContextArgs
+	 * @param array|null $aAttributes if null default to changed attributes
+	 *
+	 * @throws \ArchivedObjectException
+	 * @throws \CoreException
+	 * @throws \MissingQueryArgument
+	 * @throws \MySQLException
+	 * @throws \MySQLHasGoneAwayException
+	 * @throws \OQLException
+	 * @since 3.1.1 3.2.0 N°6228
+	 */
+	public function DoActivateForSpecificAttributes(array $aContextArgs, ?array $aAttributes)
+	{
+		$bGo = true;
+		if (isset($aContextArgs['this->object()']))
+		{
+			/** @var \DBObject $oObject */
+			$oObject = $aContextArgs['this->object()'];
+			if (is_null($aAttributes)) {
+				$bGo = $this->IsTargetObject($oObject->GetKey(), $oObject->ListPreviousValuesForUpdatedAttributes());
+			} else {
+				$aChanges = array_fill_keys($aAttributes, true);
+				$bGo = $this->IsTargetObject($oObject->GetKey(), $aChanges);
+			}
+		}
+		if ($bGo)
+		{
+			parent::DoActivate($aContextArgs);
+		}
```

Comment:
I really dislike the bGo var use :(
The dev team talked about that in the past (reducing all those unreadable if), it's far more readable to use return statements instead ! 

So I would suggest : 

```suggestion
		if (false === isset($aContextArgs['this->object()'])) {
			return;
		}
		
		/** @var \DBObject $oObject */
		$oObject = $aContextArgs['this->object()'];
		if (is_null($aAttributes)) {
			if (false === $this->IsTargetObject($oObject->GetKey(), $oObject->ListPreviousValuesForUpdatedAttributes())) {
				return;
			}
		} else {
			$aChanges = array_fill_keys($aAttributes, true);
			if (false === $this->IsTargetObject($oObject->GetKey(), $aChanges)) {
				return;
			}
		}

		/** @noinspection MissUsingParentKeywordInspection */
		parent::DoActivate($aContextArgs);
```